src/object/cli_shard.c

```diff
@@ -1691,14 +1694,17 @@ verify_csum_cb(daos_key_desc_t *kd, void *buf, unsigned int size, void *arg)
 
 		d_iov_set(&enum_type_val, buf, rec_data_len);
 
+		/* NB: csum_enum_verify_recx/sv() calls daos_csummer_verify_iod() with dummy
+		 * iod which has no akey.
+		 */
+		csummer->dcs_skip_key_calc   = true;
+		csummer->dcs_skip_key_verify = true;
 		if (kd->kd_val_type == OBJ_ITER_RECX)
-			rc = csum_enum_verify_recx(args->csummer, rec,
-						   &enum_type_val,
-						   ci_to_compare);
+			rc = csum_enum_verify_recx(csummer, rec, &enum_type_val, ci_to_compare);
 		else
-			rc = csum_enum_verify_sv(args->csummer, rec,
-						 &enum_type_val,
-						 ci_to_compare);
+			rc = csum_enum_verify_sv(csummer, rec, &enum_type_val, ci_to_compare);
+		csummer->dcs_skip_key_calc   = skip_key_calc;
+		csummer->dcs_skip_key_verify = skip_key_verify;
```

Comment:
Looks there are some other cases need be fixed as well, for example, csummer_verify_recx() in src/object/srv_enum.c.